akka-projection-testkit/src/main/scala/akka/projection/testkit/internal/TestProjectionImpl.scala

```diff
@@ -0,0 +1,202 @@
+
+import akka.Done
+import akka.actor.typed.ActorSystem
+import akka.annotation.InternalApi
+import akka.event.Logging
+import akka.event.LoggingAdapter
+import akka.projection.ProjectionId
+import akka.projection.RunningProjection
+import akka.projection.StatusObserver
+import akka.projection.internal.ActorHandlerInit
+import akka.projection.internal.HandlerStrategy
+import akka.projection.internal.InternalProjectionState
+import akka.projection.internal.OffsetStrategy
+import akka.projection.internal.ProjectionSettings
+import akka.projection.internal.RestartBackoffSettings
+import akka.projection.internal.SettingsImpl
+import akka.projection.scaladsl.SourceProvider
+import akka.projection.testkit.javadsl
+import akka.projection.testkit.scaladsl.TestOffsetStore
+import akka.projection.testkit.scaladsl.TestProjection
+import akka.stream.SharedKillSwitch
+import akka.stream.scaladsl.Source
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+private[projection] class TestProjectionImpl[Offset, Envelope] private[projection] (
+    val projectionId: ProjectionId,
+    val sourceProvider: SourceProvider[Offset, Envelope],
+    val handlerStrategy: HandlerStrategy,
+    val offsetStrategy: OffsetStrategy,
+    val statusObserver: StatusObserver[Envelope],
+    val offsetStoreFactory: () => TestOffsetStore[Offset],
+    val startOffset: Option[Offset])
+    extends TestProjection[Offset, Envelope]
+    with akka.projection.testkit.javadsl.TestProjection[Offset, Envelope]
+    with SettingsImpl[TestProjectionImpl[Offset, Envelope]] {
+
+  // singleton state is ok because restart strategies not supported.
+  // also keeps in memory offset table alive.
+  private var _state: Option[TestInternalProjectionState[Offset, Envelope]] = None
+
+  private def copy(
+      projectionId: ProjectionId = projectionId,
+      sourceProvider: SourceProvider[Offset, Envelope] = sourceProvider,
+      handlerStrategy: HandlerStrategy = handlerStrategy,
+      offsetStrategy: OffsetStrategy = offsetStrategy,
+      statusObserver: StatusObserver[Envelope] = statusObserver,
+      offsetStoreFactory: () => TestOffsetStore[Offset] = offsetStoreFactory,
+      startOffset: Option[Offset] = startOffset): TestProjectionImpl[Offset, Envelope] =
+    new TestProjectionImpl(
+      projectionId,
+      sourceProvider,
+      handlerStrategy,
+      offsetStrategy,
+      statusObserver,
+      offsetStoreFactory,
+      startOffset)
+
+  override def withStatusObserver(observer: StatusObserver[Envelope]): TestProjectionImpl[Offset, Envelope] =
+    copy(statusObserver = observer)
+
+  /**
+   * The initial offset of the offset store.
+   */
+  def withStartOffset(offset: Offset): TestProjectionImpl[Offset, Envelope] = copy(startOffset = Some(offset))
+
+  /**
+   * The offset store factory. The offset store has the same lifetime as the Projection. It is instantiated when the
+   * projection is first run and is created with [[newState]].
+   */
```

Comment:
better to not have api docs on the methods in internal

Reply:
Ok. I removed the implemented method javadocs for all internal impls.